### PR TITLE
properly null out ptr in LinkedList::split_off - fixes #26021

### LINKED ISSUE #26021: LinkedList corrupted in .split_off()
User seppo0010 in #rust irc found this corruption and crasher bug in `LinkedList::split_off()`

Exists in rustc 1.0 and nightly

[(playpen link)](https://play.rust-lang.org/?gist=2cf6a275259d981183eb&version=nightly)

```rust
use std::collections::LinkedList;

fn main() {
    let mut v1 = LinkedList::new();
    v1.push_front(1u8);
    v1.push_front(1u8);
    v1.push_front(1u8);
    v1.push_front(1u8);
    let _ = v1.split_off(3);
    assert_eq!(v1.len(), 3);
    //assert_eq!(v2.len(), 1);

    println!("{}", v1.iter().len());
    println!("{}", v1.iter().collect::<Vec<_>>().len());
}
```

Corruption is visible when you let the split off vector drop at once.

It's too late for me to analyze this right now, but I suspect a backlink is not being updated correctly.

### PR DESCRIPTION
**Pull request by Gankra**
Still running tests...

### CODE REVIEW

File: src/libcollections/linked_list.rs [hunk 1]
```diff
@@ -610,6 +610,7 @@ impl<T> LinkedList<T> {
         };
 
         mem::swap(&mut split_node.resolve().unwrap().next, &mut splitted_list.list_head);
+        splitted_list.list_head.as_mut().unwrap().prev = Rawlink::none();
```
Comment (huonw) [inline src/libcollections/linked_list.rs]: Could you add a brief comment?

**Comment by huonw** [comment]
r=me

File: src/libcollections/linked_list.rs (hunk 1 above)
Comment (Gankro) [inline src/libcollections/linked_list.rs]: What do you have in mind?

**Comment by Gankro** [comment]
whoops, the issue number is wrong.

**Comment by Gankro** [comment]
Actually fixes #26021 ...

**Comment by bluss** [comment]
Great!

**Comment by alexcrichton** [comment]
triage: beta-accepted

**Pull request merged**